packages/runtime-core/src/rendererTemplateRef.ts

```diff
@@ -61,7 +61,8 @@ export function setRef(
     return
   }
   const oldRef = oldRawRef && (oldRawRef as VNodeNormalizedRefAtom).r
-  const refs = owner.refs === EMPTY_OBJ ? (owner.refs = {}) : owner.refs
+  const refs =
+    owner.refs === EMPTY_OBJ ? (owner.refs = reactive({})) : owner.refs
```

Comment:
I don't think this is the right fix. "Options API" refs haven't been reactive in Vue 2 or Vue 3. The only way to get a reactive ref is through using a an actual `ref()`/`reactive()` object like in the reported issue.